std/algorithm/searching.d

```diff
@@ -4426,6 +4427,429 @@ private void skipAll(alias pred = "a == b", R, Es...)(ref R r, Es es)
+                Ins _needles;
+                typeof(R.init.take(0)) _skip; // skip before next hit
+                typeof(_skip) _hit; // most recent (last) hit if any
+                alias E = Tuple!(typeof(_skip), Hit);
+                alias Hit = size_t; // 0 iff no hit, otherwise hit in needles[index-1]
+                Hit _hitNr; // hit number: 0 means no hit, otherwise index+1 to needles that matched
+            }
+
+            this(R haystack, Ins needles)
+            {
+                this._rest = haystack.drop(0);
+                this._needles = needles;
+                popFront();
+            }
+
+            @property auto ref front()
+            {
+                import std.range : empty;
+                return !_skip.empty ? E(_skip, 0) : E(_hit, _hitNr);
+            }
+
+            import std.range.primitives : isInfinite;
+            static if (isInfinite!R)
+                enum empty = false; // propagate infiniteness
+            else
+                @property bool empty() const
+                {
+                    import std.range.primitives : empty;
+                    return _skip.empty && _hit.empty && _rest.empty;
+                }
+
+            void popFront() @trusted
+            {
+                import std.range.primitives : empty;
+                if (!_skip.empty)
+                {
+                    _skip = typeof(_skip).init; // jump over _skip
+                }
+                else
+                {
+                    import std.algorithm.searching : countUntil, find;
+
+                    auto match = _rest.find!pred(_needles);
+                    static if (_needles.length >= 2) // if variadic version
+                    {
+                        auto hitValue = match[0];
+                        _hitNr = match[1];
+                    }
+                    else
+                    {
+                        auto match = _rest.find!pred(_needles);
+                        auto hitValue  = needles[0];
+                        _hitNr = match.empty ? 0 : 1;
+                    }
+
+                    if (_hitNr == 0) // no more hits
+                    {
+                        _skip = _rest.take(size_t.max);
+                        _hit = typeof(_hit).init;
+                        _rest = typeof(_rest).init;
+                    }
+                    else
+                    {
+                        size_t hitLength = size_t.max;
+                        switch (_hitNr - 1)
+                        {
+                            foreach (const i; aliasSeqOf!(iota(0, n)))
+                            {
+                                case i: hitLength = _needles[i].length; break;
+                            }
+                            default:
+                                assert(hitLength < size_t.max, "hitNr should always be found");
+                        }
+
+                        const pos = _rest.countUntil(hitValue);
```

Comment:
This is the only bit that I am not happy with, but
- `find` should really allow to return the position of the hit (or the variadic version of [`countUntil` could return the matched overload](https://issues.dlang.org/show_bug.cgi?id=17634)).
- by using the generic `take`, `drop` approach `@nordlow's trick with ` _rest[0 .. hitValue.ptr - _rest.ptr]` doesn't work